source/extensions/filters/common/rbac/utility.h

```diff
@@ -32,6 +32,51 @@ namespace RBAC {
 struct RoleBasedAccessControlFilterStats {
   ENFORCE_RBAC_FILTER_STATS(GENERATE_COUNTER_STRUCT)
   SHADOW_RBAC_FILTER_STATS(GENERATE_COUNTER_STRUCT)
+
+  Stats::Scope& scope_;
+  Stats::StatName per_policy_stat_;
+  Stats::StatName per_policy_shadow_stat_;
+  Stats::StatNameSetPtr stat_name_set_;
+  const Stats::StatName unknown_policy_allowed_;
+  const Stats::StatName unknown_policy_denied_;
+  const Stats::StatName unknown_shadow_policy_allowed_;
+  const Stats::StatName unknown_shadow_policy_denied_;
+
+  void add_policy(const std::string& name) {
+    stat_name_set_->rememberBuiltin(absl::StrCat(name, ".allowed"));
+    stat_name_set_->rememberBuiltin(absl::StrCat(name, ".denied"));
+  }
+
+  void add_shadow_policy(const std::string& name) {
+    stat_name_set_->rememberBuiltin(absl::StrCat(name, ".shadow_allowed"));
+    stat_name_set_->rememberBuiltin(absl::StrCat(name, ".shadow_denied"));
+  }
+
+  void inc_policy_allowed(absl::string_view name) {
+    incCounter(per_policy_stat_,
+               stat_name_set_->getBuiltin(absl::StrCat(name, ".allowed"), unknown_policy_allowed_));
```

Comment:
Does this somehow default to `unknown`?

Reply:
The stat name for unknown metrics is defined here https://github.com/envoyproxy/envoy/pull/32313/files#diff-2dae3b425d220696313a85b77b3bf1226ddec2248705bdfb82adad8eff07f106R25

The statsd metric would look like this

```
http.<listener name>.rbac.policy.unknown_denied: 1
http.<listener name>.rbac.<optional shadow stat prefix>.policy.unknown_shadow_allowed: 1
```